### PR TITLE
Post deployment verification script

### PR DESCRIPTION
**Pull request by maurelian**
This is an early iteration on the script to validate deployments. I'd like to get it reviewed, or even merged, to validate the direction.

**How to test:** 

1. `export CONTRACTS_RPC_URL=https://kovan.infura.io/v3/YOURKEYHERE`
2. Run this:
```
npx hardhat validate:address-dictator \
--dictator 0x9816d75a8FEc3a26995aF7C1900215541Be2a210 \
--manager 0x96df6a39fB4Da2BF154e527f2Aa73B0992B15e28 \
--multisig 0x54E2e4B3A7C3045Dcf1e927b890D33BACCcCE249
```

### CODE REVIEW

File: packages/contracts/deploy/014-OVM_L1StandardBridge.deploy.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ const deployFn: DeployFunction = async (hre) => {
 
   // Check if if we're on the hardhat chain ID. This will only happen in CI. If this is the case, we
   // can skip directly to transferring ownership over to the ChugSplashDictator contract.
-  if (isHardhatNode(hre)) {
+  if (await isHardhatNode(hre)) {
```
Comment (ben-chain) [inline packages/contracts/deploy/014-OVM_L1StandardBridge.deploy.ts:89]: Were these all wrong before and using promises instead?

File: packages/contracts/tasks/validate-address-dictator.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+task('validate:address-dictator')
+  // Provided by the signature Requestor
+  .addParam(
+    'dictator',
+    'Address of the AddressDictator to validate.',
+    undefined,
+    types.string
+  )
+  // I'm not certain if this value should be entered manually or read from the artifacts,
```
Comment (ben-chain) [inline packages/contracts/tasks/validate-address-dictator.ts:34]: TODO reminder to finish this comment. I'm a proponent of entering manually, will non-deployers even have the artifacts yet?

File: packages/contracts/tasks/validate-address-dictator.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          `Deployed ${pair.name} code`,
+          artifact.deployedBytecode,
+          code
+        )
+      } else {
+        console.log(`${pair.name} not updated`)
+      }
+    }
+
+    // Verify libAddressManager is set properly
```
Comment (ben-chain) [inline packages/contracts/tasks/validate-address-dictator.ts:158]: reminder on these

**Review by ben-chain — COMMENTED** [review]
Additional validation which I can think of is:
- verify that `AM.owner == multisig.address`

File: packages/contracts/deploy/014-OVM_L1StandardBridge.deploy.ts (hunk 1 above)
Comment (smartcontracts) [reply packages/contracts/deploy/014-OVM_L1StandardBridge.deploy.ts:89]: This specific step was wrong, yes. We caught it while manually reviewing this.

File: packages/contracts/tasks/validate-address-dictator.ts (hunk 2 above)
Comment (maurelian) [reply packages/contracts/tasks/validate-address-dictator.ts:34]: No. I removed that comment.

File: packages/contracts/src/validation-utils.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  )
+}
+
+const codes = {
+  reset: '\x1b[0m',
+  red: '\x1b[0;31m',
+  green: '\x1b[0;32m',
+  cyan: '\x1b[0;36m',
+  yellow: '\x1b[1;33m',
+}
```
Comment (maurelian) [inline packages/contracts/src/validation-utils.ts:24]: Makes the wall of text easier to read.

File: packages/contracts/src/validation-utils.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      )
+    )
+    console.log(color.green(`${description} looks good! 😎`))
+  } else {
+    throw new Error(`${description} looks wrong.
+    ${expected.value}\ndoes not match\n${deployed.value}.
+    `)
+  }
+  console.log() // Add some whitespace
+}
```
Comment (maurelian) [inline packages/contracts/src/validation-utils.ts:88]: Kind of an ugly function, but it makes the scripts much cleaner.

File: packages/contracts/tasks/validate-address-dictator.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    const network = await provider.getNetwork()
+    console.log(
+      `
+Reading from the ${c.red(network.name)} network (Chain ID: ${c.red(
+        '' + network.chainId
+      )})`
+    )
+    const res = await getInput(
+      c.yellow('Please confirm that this is the correct network? (Y/n)\n> ')
```
Comment (maurelian) [inline packages/contracts/tasks/validate-address-dictator.ts]: Make this more consistent with the other prompts.

**Comment by smartcontracts** [comment]
Rebased on develop

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**